FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/internal/SplashActivity.java

```diff
@@ -1,17 +1,175 @@
 package org.firstinspires.ftc.robotcontroller.internal;
 
 import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
 import android.content.Intent;
 import android.os.Bundle;
+import android.util.Log;
 
-public class SplashActivity extends Activity {
+import com.cyanogenmod.updater.utils.MD5;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class SplashActivity extends Activity
+{
+    File libInProtectedStorage;
+    File protectedExtraFolder;
+    File libOnSdcard;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    protected void onCreate(Bundle savedInstanceState)
+    {
         super.onCreate(savedInstanceState);
 
-        Intent intent = new Intent(this, FtcRobotControllerActivity.class);
-        startActivity(intent);
-        finish();
+        if(shouldProceedWithRcLaunch())
+        {
+            /*
+             * We've been given the go-ahead! Load up libVuforiaReal.so
+             */
+            System.load(libInProtectedStorage.getAbsolutePath());
+            Log.e("lib", "loaded!");
+
+            /*
+             * Start loading the main activity. After it's loaded, close
+             * down the splash screen
+             */
+            Intent intent = new Intent(this, FtcRobotControllerActivity.class);
+            startActivity(intent);
+            finish();
+        }
+    }
+
+    private boolean shouldProceedWithRcLaunch()
+    {
+        libInProtectedStorage = new File("/data/data/com.qualcomm.ftcrobotcontroller/extra/libVuforiaReal.so");
+        protectedExtraFolder = new File("/data/data/com.qualcomm.ftcrobotcontroller/extra/");
```

Comment:
There's got to be an API to get the location of our app folder, rather than hard-coding it.